docs/content/guide/forms.ngdoc

```diff
@@ -44,7 +44,9 @@ for other directives to augment its behavior.
           };
 
           $scope.reset = function() {
-            $scope.user = angular.copy($scope.master);
+            //$scope.user = angular.copy($scope.master);
+            $scope.master = {};
+            $scope.user = {};
```

Comment:
I don't think the point of resetting is to clear the fields/data. The point is to undo any (non-committed) changes, i.e. make `user` a copy of `master`.